components/tools/ToolsDashboard.tsx

```diff
@@ -149,18 +136,30 @@ export default function ToolsDashboard() {
       }
     });
 
-    setToolsList(tempToolsList);
-  };
+    return tempToolsList;
+  }, [isPaid, isAsyncAPIOwner, languages, technologies, categories, searchName]);
+
+  // useEffect to scroll to the opened category when url has category as element id
+  useEffect(() => {
+    const { hash } = window.location;
 
+    if (hash) {
+      const elementID = decodeURIComponent(hash.slice(1));
+      const element = document.getElementById(elementID);
+
+      if (element) {
+        document.documentElement.style.scrollPaddingTop = '6rem';
+        element.scrollIntoView({ behavior: 'smooth' });
+        document.documentElement.style.scrollPaddingTop = '0';
+      }
+    }
+  }, []);
+  // Function to update the list of tools according to the current filters applied
```

Comment:
The actual logic to scroll to the element. The padding is removed afterward so that it does not interfere with the other parts of the website